src/pages/index.js

```diff
@@ -169,8 +174,7 @@ function handleAvatarSubmit({ url }) {
     });
   }
 
-  handleSubmit(makeRequest, updateAvatarModal, "Saving...").then(() => {
-    updateAvatarModal.reset();
+  handleSubmit(makeRequest, updateAvatarModal, "Saving...", "Save", () => {
     updateAvatarModal.setLoading(true, "Save");
```

Comment:
You have everything inside `handleSubmit` (the button text returns back in `finally` there)

Reply:
I believe i'd resolved my issues now. I have been struggling with my button state. I like that it goes disabled to indicate loading as well as different text for removing a card ("Removing..."). Putting the button back to enabled/disabled based on modal has been my main road block. I've swapped out the successCallback with a disableBtn. I've also added disableBtn = false to the catch error (in the event of server error, the button will not be stuck disabled. )